### ISSUE TITLE
custom info window

### ISSUE THREAD

**Issue by neo125874** [nitaliano/react-native-mapbox-gl#558]
could we custom the info window?   (the info window means the popup when annotation clicked)

**Comment by atarsha** [comment]
With custom annotations, you can create a TouchableHighlight which then shows the InfoWindow as a react-native view. You may need to play with absolute positioning to do so.

**Comment by benwalks** [comment]
@neo125874, did you get this working?  I can get custom annotations to work, but cannot get TouchableHighlights to trigger when within the Annotation component.

**Comment by neo125874** [comment]
@atarsha  thanks, i got it, but would this lib implement the callout feature like airbnb's map?

**Comment by neo125874** [comment]
@benwalks hi, i use the TouchableHighlights & the popup modal to avoid the callout offset position setting, what do you mean about not triggering?

**Comment by neo125874** [comment]
@benwalks  something like this
`<MapView>
    <Annotation>
        <View><TouchableHighlight><Image/></TouchableHighlight></View>
    </Annotation>
</MapView>`

would this help you?

**Comment by atarsha** [comment]
@benwalks I have seen behavior on Android where sometimes turning on showsUserLocation prevents the TouchableHighlight from being triggered. Try setting showsUserLocation to false and see if they start triggering.

I'm not sure what the solution is there.

As for iOS, it seems to work fine on this end.

**Comment by atarsha** [comment]
@neo125874 As for the callout feature, I don't know. I'm not a contributor to this lib, currently and I don't know what is in store for it. I do know that MapBox is not currently doing any development on this lib at the moment, and it's really up to the community to provide pull requests for them to review.

I have also found that they pay little attention to the pull requests (as evidenced by 12 stagnant pull requests, with many over 30 days old. I'm not knocking Mapbox, just trying to pass along the fact that this library is "unofficial" and doesn't get a lot of attention from Mapbox, so new features like that would be up to the community to develop, then wait for Mapbox to allocate a resource to review it.

In other words, that's probably not going to happen anytime soon.

**Comment by neo125874** [comment]
@atarsha yeah...seems like you said, maybe cuz it's not official & no other ones to take over the project...it's a pity that mapbox really a beautiful map

**Comment by benwalks** [comment]
Hi @atarsha, thanks for the tip!  Setting userTrackingMode to 'none' did the trick even though it's a less than ideal fix.

**Comment by nitaliano** [comment]
https://github.com/mapbox/react-native-mapbox-gl/pull/677
